openlane/common/drc.py

```diff
@@ -18,16 +18,20 @@
 from enum import IntEnum
 from decimal import Decimal, InvalidOperation
 from dataclasses import dataclass, field, asdict
-from typing import List, Optional, Tuple, Dict
+from typing import List, Optional, Tuple, Dict, Union
+
 
 BoundingBox = Tuple[Decimal, Decimal, Decimal, Decimal]  # microns
+BoundingBoxWithDescription = Tuple[Decimal, Decimal, Decimal, Decimal, str]  # microns
 
 
 @dataclass
 class Violation:
     rules: List[Tuple[str, str]]  # (layer, rule)
     description: str
-    bounding_boxes: List[BoundingBox] = field(default_factory=list)
+    bounding_boxes: List[Union[BoundingBox, BoundingBoxWithDescription]] = field(
```

Comment:
Sorry to be pedantic but that's still an API break. If you were iterating over `.bounding_boxes` using:

```python
for (llx, lly, urx, ury) in violation.bounding_boxes:
   pass
```

This code wouldn't work after this patch.

---

Can you reason why BoundingBoxWithDescription is needed? Can the information not be captured in `description`?

Reply:
Why won't the code work? If `violations.bounding_boxes` is of type `BoundingBox` code will still work.

`description` is common for all bounding boxes for a certain category. To add the additional information in the image below you need a text field with the bounding box.

![image](https://github.com/user-attachments/assets/29ff21f0-ac78-477f-829a-d072c089b34e)

Reply:
I made another update. I think it is more reasonable.